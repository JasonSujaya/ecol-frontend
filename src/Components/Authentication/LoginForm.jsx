import React from "react";

class LoginForm extends React.Component {
  render() {
    return <div>Login stuff here</div>;
  }
}

export default LoginForm;
