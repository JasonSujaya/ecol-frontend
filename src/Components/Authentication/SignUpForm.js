import React from "react";
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";
import "./authentication.css";

import {
  signUp,
  login,
  inputNotEmpty,
  checkPassword,
} from "./authentication.js";

class SignUpForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      first_name: "",
      last_name: "",
      password: "",
      validForm: false,
      invalidPassword: false,
      invalidPasswordMessage: "",
      error: false,
      signUpSuccessfull: false,
    };
  }

  onSignUp = async (e) => {
    e.preventDefault();
    this.setState({
      invalidPassword: true,
      invalidPasswordMessage: checkPassword(this.state.password),
    });

    if (checkPassword(this.state.password) == true) {
      let result = signUp(
        this.state.email,
        this.state.first_name,
        this.state.last_name,
        this.state.password
      );
      result
        .then((response) => {
          let result = login(this.state.email, this.state.password);
          result.then((response) => {
            this.props.onSuccess(response.data);
            this.setState({
              email: "",
              first_name: "",
              last_name: "",
              password: "",
            });
          });
        })
        .catch((error) => {
          this.setState({ error: true });
        });
    } else {
      this.setState({
        invalidPassword: true,
        invalidPasswordMessage: checkPassword(this.state.password),
      });
    }
  };

  checkValidForm = (e) => {
    let validity =
      inputNotEmpty(this.state.email) &&
      inputNotEmpty(this.state.password) &&
      inputNotEmpty(this.state.first_name) &&
      inputNotEmpty(this.state.last_name);
    this.setState({ validForm: validity });
  };

  onErrorDisplayMessage = () => {
    if (this.state.error) {
      return (
        <div className="signUpError">You have an existing account already!</div>
      );
    }
  };

  invalidPasswordDisplayMessage = () => {
    if (this.state.invalidPasswordMessage) {
      return (
        <div className="invalidPasswordError">
          {this.state.invalidPasswordMessage}
        </div>
      );
    }
  };

  render() {
    return (
      <Grid container direction="column" justify="center" alignItems="center">
        <form noValidate autoComplete="off">
          <div className="formContainer">
            <TextField
              required
              className="emailInput"
              id="standard-required"
              label="email"
              onChange={(e) =>
                this.setState(
                  {
                    email: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.email}
            />
            <TextField
              id="standard-password-input"
              className="passwordInput"
              label="password"
              type="password"
              required
              onChange={(e) =>
                this.setState(
                  {
                    password: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.password}
            />
            <TextField
              required
              id="standard-required"
              className="firstNameInput"
              label="Your First Name"
              onChange={(e) =>
                this.setState(
                  {
                    first_name: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.first_name}
            />
            <TextField
              required
              id="standard-required"
              label="Your Last Name"
              className="lastNameInput"
              onChange={(e) =>
                this.setState(
                  {
                    last_name: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.last_name}
            />
            <Button
              variant="contained"
              onClick={this.onSignUp}
              disabled={!this.state.validForm}
              className="signUpButton"
            >
              Sign Up
            </Button>
          </div>
        </form>
        {this.onErrorDisplayMessage()}
        {this.invalidPasswordDisplayMessage()}
      </Grid>
    );
  }
}

export default SignUpForm;
