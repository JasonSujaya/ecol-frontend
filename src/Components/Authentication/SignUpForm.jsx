import React from "react";
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";
import { signUp, inputNotEmpty, checkPassword } from "./authentication.js";

const styles = (theme) => ({});

class SignUpForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      first_name: "",
      last_name: "",
      password: "",
      validForm: false,
    };
  }

  onSignUp = (e) => {
    e.preventDefault();
    if (checkPassword(e)) {
      signUp(
        this.state.email,
        this.state.first_name,
        this.state.last_name,
        this.state.password
      );
    } else {
      console.log(checkPassword(e));
    }
    this.setState({ email: "", first_name: "", last_name: "", password: "" });
  };

  checkValidForm = (e) => {
    let validity =
      inputNotEmpty(this.state.email) &&
      inputNotEmpty(this.state.password) &&
      inputNotEmpty(this.state.first_name) &&
      inputNotEmpty(this.state.last_name);
    this.setState({ validForm: validity });
  };

  render() {
    return (
      <Grid container direction="column" justify="center" alignItems="center">
        <form noValidate autoComplete="off">
          <div>
            <TextField
              required
              className="emailInput"
              id="standard-required"
              label="email"
              onChange={(e) =>
                this.setState(
                  {
                    email: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.email}
            />
            <TextField
              id="standard-password-input"
              className="passwordInput"
              label="password"
              type="password"
              required
              onChange={(e) =>
                this.setState(
                  {
                    password: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.password}
            />
            <TextField
              required
              id="standard-required"
              className="firstNameInput"
              label="Your First Name"
              onChange={(e) =>
                this.setState(
                  {
                    first_name: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.first_name}
            />
            <TextField
              required
              id="standard-required"
              label="Your Last Name"
              className="lastNameInput"
              onChange={(e) =>
                this.setState(
                  {
                    last_name: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.last_name}
            />
            <Button
              variant="contained"
              onClick={this.onSignUp}
              disabled={!this.state.validForm}
              className="signUpButton"
            >
              Sign Up
            </Button>
          </div>
        </form>
      </Grid>
    );
  }
}

export default SignUpForm;
