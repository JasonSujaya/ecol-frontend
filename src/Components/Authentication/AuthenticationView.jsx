import React from "react";
import LoginForm from "./LoginForm";
import SignUpForm from "./SignUpForm";
import Button from "@material-ui/core/Button";
import { storeTokenId } from "./authentication.js";
import { withRouter } from "react-router";

class AuthenticationView extends React.Component {
  constructor(props) {
    super(props);
    this.state = { display: "LogIn" };
  }

  showAuthDisplay = () => {
    if (this.state.display == "LogIn") {
      return (
        <div>
          <LoginForm
            id="loginForm"
            onSuccess={this.onSuccesfullAuthentication}
          ></LoginForm>
          <div>
            Don't have an account yet?
            <Button
              onClick={() => {
                this.changeDisplay("SignUp");
              }}
            >
              SignUp
            </Button>
          </div>
        </div>
      );
    } else if (this.state.display == "SignUp") {
      return (
        <div>
          <SignUpForm
            id="signUpForm"
            onSuccess={this.onSuccesfullAuthentication}
          ></SignUpForm>
          <div>
            Already registered?
            <Button
              onClick={() => {
                this.changeDisplay("LogIn");
              }}
            >
              Login
            </Button>
          </div>
        </div>
      );
    }
  };

  changeDisplay = (type) => {
    this.setState({ display: type });
  };

  onSuccesfullAuthentication = (getData) => {
    console.log(getData);
    console.log("Triggered");

    storeTokenId(getData);

    this.props.onLogin(getData);
    this.props.history.push("/");
  };

  render() {
    return <div>{this.showAuthDisplay()}</div>;
  }
}

export default withRouter(AuthenticationView);
