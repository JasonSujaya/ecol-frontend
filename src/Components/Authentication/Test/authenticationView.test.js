import { createShallow, createMount } from "@material-ui/core/test-utils";
import LoginForm from "../LoginForm";
import SignUpForm from "../SignUpForm";
import AuthenticationView from "../AuthenticationView";

describe("Sign In component", () => {
  let shallow;
  let mount;

  beforeAll(() => {
    shallow = createShallow({ dive: true });
    mount = createMount({ dive: true });
  });

  it("Authentication renders correctly", () => {
    const wrapper = mount(<AuthenticationView />);
    expect(wrapper).toMatchSnapshot();
  });

  it("When display state is changes, expect to find the correct component", () => {
    //Act
    const wrapper = mount(<AuthenticationView />);
    wrapper.setState({ display: "LogIn" });

    // Assert
    expect(wrapper.contains(<LoginForm />)).toBe(true);
    expect(wrapper.contains(<SignUpForm />)).toBe(false);
    wrapper.setState({ display: "SignUp" });
    expect(wrapper.contains(<LoginForm />)).toBe(false);
    expect(wrapper.contains(<SignUpForm />)).toBe(true);
  });
});
