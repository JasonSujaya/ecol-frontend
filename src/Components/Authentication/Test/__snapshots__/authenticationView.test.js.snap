// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In component Authentication renders correctly 1`] = `ReactWrapper {}`;
